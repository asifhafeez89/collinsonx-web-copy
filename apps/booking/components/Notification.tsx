import { Box, Flex, Text } from '@collinsonx/design-system/core';
import { AlertIcon } from '@collinsonx/design-system/assets/icons';
import { ReactNode } from 'react';

interface NotificationProps {
  children: ReactNode;
}

function Notification({ children }: NotificationProps) {
  return (
    <Box
      p={16}
      style={{
        backgroundColor: 'rgba(212, 42, 84, 0.1)',
        border: '2px solid #D42A54',
        borderRadius: 8,
      }}
    >
      <Flex
        direction="row"
        justify="flex-start"
        gap="4px"
        style={{ height: '100%' }}
      >
        <Box
          sx={{
            '@media (max-width: 768px)': {
              paddingTop: '0.2rem',
            },
          }}
        >
          <AlertIcon style={{ width: 24, height: 24 }} />
        </Box>

        <Text
          size={16}
          align="left"
          fw={400}
          style={{ wordWrap: 'break-word' }}
        >
          {children}
        </Text>
      </Flex>
    </Box>
  );
}

export default Notification;
