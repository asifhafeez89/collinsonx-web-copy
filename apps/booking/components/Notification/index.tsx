import { Box, Flex, Text } from '@collinsonx/design-system/core';
import { AlertIcon } from '@collinsonx/design-system/assets/icons';
import { ReactNode } from 'react';

import classes from './Notification.module.css';

interface NotificationProps {
  children: ReactNode;
}

function Notification({ children }: NotificationProps) {
  return (
    <Box p={16} className={classes.container}>
      <Flex direction="row" justify="flex-start" gap="4px">
        <Box className={classes.iconContainer}>
          <AlertIcon style={{ width: 24, height: 24 }} />
        </Box>

        <Text fw={400} className={classes.text}>
          {children}
        </Text>
      </Flex>
    </Box>
  );
}

export default Notification;
