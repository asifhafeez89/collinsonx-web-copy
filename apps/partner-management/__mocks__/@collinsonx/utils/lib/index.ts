export const setItem = () => {
  return [];
};

export const removeItem = () => [];
