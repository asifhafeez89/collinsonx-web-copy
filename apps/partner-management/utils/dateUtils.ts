import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

// Extend dayjs with the necessary plugins
dayjs.extend(utc);
dayjs.extend(timezone);

/**
 * Formats a date string into a specified format.
 * @param dateString The date string to format.
 * @param formatString The format to apply. Default is 'DD MMM YYYY HH:mm [GMT]'.
 * @param timeZone The time zone to use for formatting. Default is 'GMT'.
 * @returns The formatted date string or an error message if the input is invalid.
 */
export function formatDateString(
  dateString: string,
  formatString: string = 'DD MMM YYYY HH:mm [GMT]',
  timeZone: string = 'GMT'
): string {
  try {
    const date = dayjs(dateString);
    if (!date.isValid()) {
      return '';
    }

    return date.tz(timeZone).format(formatString);
  } catch (error) {
    console.error('Error formatting date:', error);
    return '';
  }
}
