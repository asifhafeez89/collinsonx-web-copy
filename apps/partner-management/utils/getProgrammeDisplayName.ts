export const getProgrammeDisplayName = (programme: string): string => {
  const displayNames: { [key: string]: string } = {
    LK: 'LoungeKey',
    LP: 'LoungePass',
    PP: 'Priority Pass',
  };

  return displayNames[programme] || programme;
};
