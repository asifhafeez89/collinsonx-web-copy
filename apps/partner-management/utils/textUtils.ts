export function toTitleCase(string: string): string {
  return string.toLowerCase().replace(/^\w/, (c) => c.toUpperCase());
}
