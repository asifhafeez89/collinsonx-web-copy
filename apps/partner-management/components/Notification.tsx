import {
  Pending as IconPending,
  Confirmed as IconConfirmed,
  Declined as IconDeclined,
} from '@collinsonx/design-system/assets/icons';
import { Box, Flex, Text } from '@collinsonx/design-system/core';
import { BookingStatus } from '@collinsonx/utils';
import { bookingConfig } from 'config/booking';

const { Initialized, Confirmed, Declined, Cancelled, CheckedIn } =
  BookingStatus;

export interface NotificationProps {
  type: BookingStatus;
  children: string | JSX.Element;
}

const Notification = ({ type, children }: NotificationProps) => {
  return (
    <Box w="100%" px={40} py={16} bg={bookingConfig[type].color}>
      <Flex gap={8} align="center">
        {type === Initialized && <IconPending />}
        {(type === Confirmed || type === CheckedIn) && <IconConfirmed />}
        {type === Declined && <IconDeclined />}
        {type === Cancelled && <IconDeclined />}
        <Text>{children}</Text>
      </Flex>
    </Box>
  );
};
export default Notification;
