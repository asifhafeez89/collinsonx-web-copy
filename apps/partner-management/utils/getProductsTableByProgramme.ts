import { AncillaryProduct, Maybe, Product } from '@collinsonx/utils';
import { GroupedProducts } from 'types/GroupedProducts';
import formatCurrency from '@automattic/format-currency';

const notNull = (
  value: Maybe<Product> | Maybe<AncillaryProduct>
): value is Product | AncillaryProduct => !!value;

const isProduct = (value: Product | AncillaryProduct): value is Product =>
  value.hasOwnProperty('category');

export const getProductsTableByProgramme = (
  products: Array<Maybe<Product> | Maybe<AncillaryProduct>>
): GroupedProducts => {
  return products
    .filter(notNull)
    .flatMap((product) =>
      product.costs.map((cost) => ({
        ...product,
        cost,
        category: isProduct(product) ? product.category : 'Ancillary Product',
      }))
    )
    .reduce((groupedProducts: GroupedProducts, product) => {
      const { name, cost, salePrices, status, tier } = product;
      const programme = cost?.programme;

      if (!programme) {
        return groupedProducts;
      }

      groupedProducts[programme] = groupedProducts[programme] || [];
      groupedProducts[programme].push({
        category: product.category,
        name,
        tier: tier || 'N/A',
        status,
        salePrice: formatCurrency(
          salePrices[0]?.salePrice || 0,
          salePrices[0]?.salePriceCurrency || ''
        ),
        cost: formatCurrency(cost.cost || 0, cost.costCurrency || ''),
        tax: `${cost.defaultTaxPercentage}%`,
        costType: cost.type,
      });

      return groupedProducts;
    }, {});
};
