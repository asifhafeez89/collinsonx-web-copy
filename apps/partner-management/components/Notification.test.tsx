import Notification from './Notification';
import { BookingStatus } from '@collinsonx/utils';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@collinsonx/design-system/test-utils';

describe('<Notification />', () => {
  it('should render', () => {
    const component = render(
      <Notification type={BookingStatus.Pending}>a</Notification>
    );

    expect(component).toMatchSnapshot();
  });
});
