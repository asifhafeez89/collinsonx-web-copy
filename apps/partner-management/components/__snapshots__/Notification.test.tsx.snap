// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  padding-right: 2.5rem;
  padding-left: 2.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  background: #FFF3BF;
  width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-3:focus:not(:focus-visible) {
  outline: none;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2 emotion-3"
          >
            a
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  padding-right: 2.5rem;
  padding-left: 2.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  background: #FFF3BF;
  width: 100%;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-3:focus:not(:focus-visible) {
  outline: none;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2 emotion-3"
        >
          a
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
